データベース設計

RyuseiNono opened the pull request:
# ER図
データベース設計のため作成しました。
https://gyazo.com/0f8ce4434c81e60928b4904048b739a6

# README
テーブルの内容とアソシエーションについて記載しました。

RyuseiNono committed:
Basic認証導入

RyuseiNono committed:
Basic認証不使用設定

kokikurahashi commented:
レビュー依頼ありがとうございます。
READMEに関する変更点がないためコミットプッシュのし忘れをご確認ください。
またREADMEに記載をしていない場合は、DB設計をREADMEに記載をお願いします。

RyuseiNono committed:
README作成

RyuseiNono commented:
コミットし忘れていました。
コミットプッシュを行いましたので確認よろしくお願いします。

kuu0123 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| ------------- | ---------- | ------------------------------ |
+| name          | string     | null: false                    |
+| info          | text       | null: false                    |
+| status        | integer    | null: false                    |
+| category      | integer    | null: false                    |
+| shipping-cost | integer    | null: false                    |
+| ship-from     | integer    | null: false                    |
+| days_to_ship  | integer    | null: false                    |
```
```ruby
| status        | integer    | null: false                    |
| category      | integer    | null: false                    |
| shipping-cost | integer    | null: false                    |
| ship-from     | integer    | null: false                    |
| days_to_ship  | integer    | null: false                    |
```
商品の状態, 配送料の負担, 発送元の地域,  発送までの日数, カテゴリーのカラムはactive_hashで実装の予定のため、単数形＋「_id」にしましょう。
現状のカラム名だと予期せぬエラーをまねく可能性があります。

また、29,30行目のように複数の単語をつなげる場合は、ハイフンではなくアンダースコアでつなぐようにしましょう。

kuu0123 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| ship-from     | integer    | null: false                    |
+| days_to_ship  | integer    | null: false                    |
+| price         | integer    | null: false                    |
+| user_id       | references | null: false, foreign_key: true |
```
```ruby
	| user_id       | references | null: false, foreign_key: true |
```
references 型で記述する際は、カラム名の_idは不要となります。
もしカラム名で_idを使用する予定であれば、integer型を使用した方が良いでしょう。

kuu0123 commented on README.md:
```diff
@@ -1,24 +1,67 @@
-* Deployment instructions
+| Column  | Type           | Options                        |
+| ------- | -------------- | ------------------------------ |
+| user_id | dareferenceste | null: false, foreign_key: true |
+| item_id | references     | null: false, foreign_key: true |
```
```ruby
| user_id | dareferenceste | null: false, foreign_key: true |
| item_id | references     | null: false, foreign_key: true |
```
references 型で記述する際は、カラム名の_idは不要となります。
もしカラム名で_idを使用する予定であれば、integer型を使用した方が良いでしょう。

kuu0123 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| street_address | string     | null: false                    |
+| building       | string     |                                |
+| phone_number   | integer    | null: false                    |
+| order_id       | references | null: false, foreign_key: true |
```
```ruby
	| order_id       | references | null: false, foreign_key: true |
```
references 型で記述する際は、カラム名の_idは不要となります。
もしカラム名で_idを使用する予定であれば、integer型を使用した方が良いでしょう。

kuu0123 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| Column         | Type       | Options                        |
+| -------------- | ---------- | ------------------------------ |
+| postal_code    | integer    | null: false                    |
+| state          | integer    | null: false                    |
```
```ruby
	| state          | integer    | null: false                    |
```
```ruby
	| prefecture_id | integer    | null: false                    |
```
都道府県に関しては、出品時に使用した「発送元の地域」と同じactive hashを使用したいと考えています。
そのため、カラム名も統一しておくといいでしょう。
どちらに合わせても問題ありませんが、「prefecture_id」に合わせた方が、「都道府県」という情報を取り扱っているとわかりやすいかもしれません。

kuu0123 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| city           | string     | null: false                    |
+| street_address | string     | null: false                    |
+| building       | string     |                                |
+| phone_number   | integer    | null: false                    |
```
```ruby
	| phone_number   | integer    | null: false                    |
```
電話番号は数字の文字列として実装しておきたいためstring型で実装しましょう。
integer型で実装すると、DB保存時に先頭の0が削除されてしまいます。

RyuseiNono replied:
# What
指摘箇所を修正し、READMEを再度PUSHしました
postal_codeをstringで定義しました

#Why
今回指摘いただいたphone_numberと同様にpostal_codeも先頭に0が来る可能性があるため

kuu0123 reviewed (commented):
レビュー依頼およびコミットプッシュ、ありがとうございます！
コメントしましたので、ご確認をお願いいたします。

RyuseiNono committed:
README修正

Associationにて定義するものの名前には"_id"は不要である。
電話番号などの０から始まる可能性のあるものはintegerで定義すると先頭の０が消えるためstringで定義した。

masahirokurauchi reviewed (commented):
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！

RyuseiNono committed:
微修正
